### PR TITLE
[Java] Adds extra providers

### PR DESCRIPTION
**Pull request by Craigacp**
**Description**:
Adds hooks for using ArmNN, CoreML and ROCM to the Java API.

Also adds an OrtFlags mechanism for aggregating bitset enums on the Java side, and updates the NNAPI call to use them (along with the new CoreML one).

**Motivation and Context**
- Why is this change required? What problem does it solve? C API <-> Java API parity.

Depends on #6765 as it includes the flag added there.

There are other updates necessary to get the Java API back to parity in terms of the providers, but I'm waiting on responses to #6553 and #6676 before I can make progress.

### CODE REVIEW

File: java/src/main/java/ai/onnxruntime/providers/CoreMLFlags.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * Licensed under the MIT License.
+ */
+package ai.onnxruntime.providers;
+
+/** Flags for the CoreML provider. */
+public enum CoreMLFlags implements OrtFlags {
+  CPU_ONLY(1), // COREML_FLAG_USE_CPU_ONLY(0x001)
```
Comment (yuslepukhin) [inline java/src/main/java/ai/onnxruntime/providers/CoreMLFlags.java:9]: >CPU_ONLY(1) [](start = 2, length = 11)

I wonder if EnumSet can be used for validation of the flags?

Comment (Craigacp) [reply java/src/main/java/ai/onnxruntime/providers/CoreMLFlags.java:9]: What kind of validation?

Comment (yuslepukhin) [reply java/src/main/java/ai/onnxruntime/providers/CoreMLFlags.java:9]: Valid flag values

---
In reply to: [581338623](https://github.com/microsoft/onnxruntime/pull/6770#discussion_r581338623) [](ancestors = 581338623)

Comment (Craigacp) [reply java/src/main/java/ai/onnxruntime/providers/CoreMLFlags.java:9]: It only accepts valid enum constants of that type, and ensures that at most one of each enum constant is present in the set, so it's already doing that.

File: java/src/main/native/ai_onnxruntime_OrtSession_SessionOptions.c [hunk 2]
```diff
@@ -502,3 +505,53 @@ JNIEXPORT void JNICALL Java_ai_onnxruntime_OrtSession_00024SessionOptions_addACL
...
+ * Class:     ai_onnxruntime_OrtSession_SessionOptions
+ * Method:    addCoreML
+ * Signature: (JJI)V
+ */
+JNIEXPORT void JNICALL Java_ai_onnxruntime_OrtSession_00024SessionOptions_addCoreML
+  (JNIEnv * jniEnv, jobject jobj, jlong apiHandle, jlong handle, jint coreMLFlags) {
+    (void)jobj;
+  #ifdef USE_CORE_ML
+    checkOrtStatus(jniEnv,(const OrtApi*)apiHandle,OrtSessionOptionsAppendExecutionProvider_CoreML((OrtSessionOptions*) handle, (uint32_t) coreMLFlags));
+ORT_API_STATUS(OrtSessionOptionsAppendExecutionProvider_ROCM, _In_ OrtSessionOptions* options, int device_id, size_t hip_mem_limit);
```
Comment (gwang-msft) [inline java/src/main/native/ai_onnxruntime_OrtSession_SessionOptions.c:535]: OrtSessionOptionsAppendExecutionProvider_CoreML

Comment (Craigacp) [reply java/src/main/native/ai_onnxruntime_OrtSession_SessionOptions.c:535]: Ooops, yeah I'll fix that.

Comment (Craigacp) [reply java/src/main/native/ai_onnxruntime_OrtSession_SessionOptions.c]: Done.

**Review by yuslepukhin — APPROVED** [review]
:shipit:

**Pull request merged**